src/stream/successors.rs

```diff
@@ -0,0 +1,113 @@
+use std::pin::Pin;
+use std::mem;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll, ready};
+
+use pin_project_lite::pin_project;
+
+
+
+pin_project! {
+    /// A stream that yields elements by calling an async closure with the previous value as an
+    /// argument
+    ///
+    /// This stream is constructed by [`successors`] function
+    ///
+    /// [`succcessors`]: fn.succssors.html
+    #[cfg(feature = "unstable")]
+    #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+    #[derive(Debug)]
+    pub struct Successors<F, Fut, T>
+    where
+        Fut: Future<Output = Option<T>>,
+    {
+        succ: F,
+        #[pin]
+        future: Option<Fut>,
+        slot: Option<T>,
+    }
+}
+
+/// Creates a new stream where to produce each new element a closure is called with the previous
+/// value.
+///
+/// #Examples
+///
+/// ```
+/// # fn main() { async_std::task::block_on(async {
+/// #
+/// use async_std::prelude::*;
+/// use async_std::stream;
+///
+/// let s = stream::successors(Some(22), |val| {
+///     async move {
+///         Some(val + 1)
+///     }
+/// });
+///
+/// pin_utils::pin_mut!(s);
+/// assert_eq!(s.next().await, Some(22));
+/// assert_eq!(s.next().await, Some(23));
+/// assert_eq!(s.next().await, Some(24));
+/// assert_eq!(s.next().await, Some(25));
+///
+///
+///let never = stream::successors(None, |val: usize| {
+///     async move {
+///         Some(val + 1)
+///     }
+/// });
+///
+/// pin_utils::pin_mut!(never);
+/// assert_eq!(never.next().await, None);
+/// assert_eq!(never.next().await, None);
+/// #
+/// # }) }
+///
+/// ```
+#[cfg(feature = "unstable")]
+#[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+pub fn successors<F, Fut, T>(first: Option<T>, succ: F) -> Successors<F, Fut, T>
+where
+    F: FnMut(T) -> Fut,
+    Fut: Future<Output = Option<T>>,
+    T: Copy,
```

Comment:
Do need a copy trait? It seems too restrictive to me.

Reply:
I managed to remove the trait bound. The FnMut was also wrong, as the one in std takes a &T and my version to a T. 
This should be up to date now.